src/table/node-data-cell.tsx

```diff
@@ -1,54 +1,46 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 import clsx from 'clsx';
-import React, { useEffect, useState } from 'react';
+import React, { useRef } from 'react';
 import InternalStatusIndicator from '../status-indicator/internal';
 import { supportsStickyPosition } from '../internal/utils/dom';
 import styles from './styles.css.js';
 import LiveRegion from '../internal/components/live-region';
-import { TableProps } from './interfaces';
+import { useResizeObserver } from '@cloudscape-design/component-toolkit/internal';
 
 interface NoDataCellProps {
-  variant: TableProps.Variant;
-  containerWidth: number;
   totalColumnsCount: number;
   hasFooter: boolean;
   loading?: boolean;
   loadingText?: string;
   empty?: React.ReactNode;
   tableRef: React.RefObject<HTMLTableElement>;
+  containerRef: React.RefObject<HTMLElement>;
 }
 
 export function NoDataCell({
-  variant,
-  containerWidth,
   totalColumnsCount,
   hasFooter,
   loading,
   loadingText,
   empty,
   tableRef,
+  containerRef,
 }: NoDataCellProps) {
-  const [tablePaddings, setTablePaddings] = useState(containerWidth);
+  const cellContentRef = useRef<HTMLDivElement>(null);
 
-  useEffect(() => {
-    if (tableRef.current) {
+  useResizeObserver(containerRef, ({ contentBoxWidth }) => {
```

Comment:
For a better performance we probably should attach only one observer for the ref (because each observation callback is a separate react rendering loop).

Fine by now, because there is no react state inside, but this is something to keep an eye for the future